src/gui/tray/activitylistmodel.cpp

```diff
@@ -542,11 +514,145 @@ void ActivityListModel::activitiesReceived(const QJsonDocument &json, int status
     emit activityJobStatusCode(statusCode);
 }
 
+std::pair<int, int> ActivityListModel::rowRangeForEntryType(const ActivityEntryType type)
+{
+    // We want to present activities in a certain order, and we want to ensure entry types are grouped together.
+    // We therefore need to find the range of rows in the finalList that represent an entry type group.
+    int startRow = 0;
+
+    // Start from the type that we want to push down the furthest. Cascade through switch cases.
+    switch(type) {
+    case ActivityEntryType::MoreActivitiesAvailableType:
+        startRow = _finalList.count();
+        break;
+    case ActivityEntryType::ActivityType:
+        startRow += _syncFileItemLists.count();
+    case ActivityEntryType::SyncFileItemType:
+        startRow += _notificationLists.count();
+    case ActivityEntryType::NotificationType:
+        // We only show one activity for ignored files
+        if(_listOfIgnoredFiles.count() > 0) {
+            startRow += 1;
+        }
+    case ActivityEntryType::IgnoredFileType:
+        startRow += _notificationErrorsLists.count();
+    // Remaining types should go at top
+    default:
+        break;
```

Comment:
same comment
we should explicitly list all enum values to allow compiler detect that future new values are not handled

Reply:
Removed